packages/components/src/panel/test/body.js

```diff
@@ -1,97 +1,126 @@
-			expect( panelBody.text() ).toBe( 'Some Text' );
+			expect( panelContent ).toBeTruthy();
 		} );
 
-		it( 'should pass children prop but not render when sidebar is closed', () => {
-			const panelBody = shallow(
-				<PanelBody children="Some Text" initialOpen={ false } />
+		it( 'should render as initially opened, if specified', () => {
+			const { container } = render(
+				<PanelBody initialOpen={ true }>
+					<div>Content</div>
+				</PanelBody>
 			);
-			expect( panelBody.instance().props.children ).toBe( 'Some Text' );
-			// Text should be empty even though props.children is set.
-			expect( panelBody.text() ).toBe( '' );
+			const panelContent = getPanelBodyContent( container );
+
+			expect( panelContent ).toBeTruthy();
 		} );
 	} );
 
-	describe( 'mounting behavior', () => {
-		it( 'should mount with a default of being opened', () => {
-			const panelBody = mount( <PanelBody /> );
-			expect( panelBody.state( 'opened' ) ).toBe( true );
-		} );
+	describe( 'toggling', () => {
+		it( 'should toggle collapse with opened prop', () => {
+			const { container, rerender } = render(
+				<PanelBody opened={ true }>
+					<div>Content</div>
+				</PanelBody>
+			);
+			let panelContent = getPanelBodyContent( container );
 
-		it( 'should mount with a state of not opened when initialOpen set to false', () => {
-			const panelBody = mount( <PanelBody initialOpen={ false } /> );
-			expect( panelBody.state( 'opened' ) ).toBe( false );
-		} );
-	} );
+			expect( panelContent ).toBeTruthy();
+
+			act( () => {
+				rerender(
+					<PanelBody opened={ false }>
+						<div>Content</div>
+					</PanelBody>
+				);
+			} );
 
-	describe( 'toggling behavior', () => {
-		const fakeEvent = { preventDefault: () => undefined };
+			panelContent = getPanelBodyContent( container );
 
-		it( 'should set the opened state to false when a toggle fires', () => {
-			const panelBody = mount( <PanelBody /> );
-			panelBody.instance().toggle( fakeEvent );
-			expect( panelBody.state( 'opened' ) ).toBe( false );
+			expect( panelContent ).toBeFalsy();
+
+			act( () => {
+				rerender(
+					<PanelBody opened={ true }>
+						<div>Content</div>
+					</PanelBody>
+				);
+			} );
+
+			panelContent = getPanelBodyContent( container );
+
+			expect( panelContent ).toBeTruthy();
 		} );
 
-		it( 'should set the opened state to true when a toggle fires on a closed state', () => {
-			const panelBody = mount( <PanelBody initialOpen={ false } /> );
-			panelBody.instance().toggle( fakeEvent );
-			expect( panelBody.state( 'opened' ) ).toBe( true );
+		it( 'should toggle when clicking header', () => {
+			const { container } = render(
+				<PanelBody title="Panel" initialOpen={ false }>
+					<div>Content</div>
+				</PanelBody>
+			);
+			let panelContent = getPanelBodyContent( container );
+			const panelToggle = getPanelToggle( container );
+
+			expect( panelContent ).toBeFalsy();
+
+			act( () => {
+				fireEvent.click( panelToggle );
+			} );
```

Comment:
React will automatically flush effects when events are dispatched, so there's no need to use `act` here too. I'm not completely sure about this though.